pkg/storage/fs/owncloud/owncloud.go

```diff
@@ -1526,15 +1526,19 @@ func (fs *ocfs) trash(ctx context.Context, ip string, rp string, origin string)
 	// move to trash location
 	dtime := time.Now().Unix()
 	tgt := filepath.Join(rp, fmt.Sprintf("%s.d%d", filepath.Base(ip), dtime))
+
+	// The condition reads: "if the file exists"
+	// I know this check is hard to read because of the double negation
+	// but this way we avoid to duplicate the code following the if block.
+	// If two deletes happen fast consecutively they will have the same `dtime`,
+	// therefore we have to increase the 'dtime' to avoid collisions.
+	if _, err := os.Stat(tgt); !errors.Is(err, os.ErrNotExist) {
+		// timestamp collision, try again with higher value:
+		dtime++
+		tgt = filepath.Join(rp, fmt.Sprintf("%s.d%d", filepath.Base(ip), dtime))
```

Comment:
This is a good improvement, so IMO we should have it.
But IMO I can write a test where there are lots of `textfile.txt` in different folders. And I delete then all quickly. The first 2 will delete fine. Then, if the deletes happen more than once per second, there will quickly be a case where the file "exists" in the trashbin with the current Unix second, and one exists for the next Unix second, so the 3rd (or 4th and/or...)

We have this sort of stuff in oC10 where trashbin and versions entries have uses of Unix second in the back-end implementation. And that means there are timing issues when multiple deletes, versions,... happen in the same second. And these things can happen, specially when a sync-client "wakes up" and has a backlog of things to do, or if some automated data-producer is doing "lots of stuff fast".

Reply:
I completely agree but I am kind of hesitant to spend too much time on the owncloud storage driver since we don't really use it anywhere except for the tests and I still don't know if we even intend to use it productively. Maybe we need to bring this question up again in the standup or some other discussion...

Reply:
I agree - decisions are needed about which storage drivers are to be "long-term" and enhanced to pass "all the tests".